lib/src/Uncertainty/Distribution/BlockIndependentDistribution.cxx

```diff
@@ -0,0 +1,742 @@
+  if ((conditioningDimension == 0) || (hasIndependentCopula())) return getMarginal(y.getDimension()).computePDF(x);
+  // General case
+  UnsignedInteger distributionIndex = 0;
+  UnsignedInteger partialDimension = distributionCollection_[distributionIndex].getDimension();
+  while (partialDimension < conditioningDimension)
+  {
+    ++distributionIndex;
+    partialDimension += distributionCollection_[distributionIndex].getDimension();
+  }
+  // Extract the relevant part of the conditioning
+  const UnsignedInteger conditioningSize = partialDimension - conditioningDimension;
+  Point conditioningVector(conditioningSize);
+  for (UnsignedInteger i = 0; i < conditioningSize; ++i) conditioningVector[i] = y[conditioningDimension - conditioningSize + i];
+  return distributionCollection_[distributionIndex].computeConditionalPDF(x, conditioningVector);
+}
+
+Point BlockIndependentDistribution::computeSequentialConditionalPDF(const Point & x) const
+{
+  if (x.getDimension() != dimension_) throw InvalidArgumentException(HERE) << "Error: cannot compute sequential conditional PDF with an argument of dimension=" << x.getDimension() << " different from distribution dimension=" << dimension_;
+  Point result(dimension_);
+  if (hasIndependentCopula())
+    for (UnsignedInteger i = 0; i < dimension_; ++i)
+      result[i] = getMarginal(i).computePDF(x[i]);
+  else
+    {
+      const UnsignedInteger size = distributionCollection_.getSize();
+      UnsignedInteger start = 0;
+      UnsignedInteger stop = 0;
+      for (UnsignedInteger i = 0; i < size; ++i)
+	{
+	  const UnsignedInteger localDimension = distributionCollection_[i].getDimension();
+	  Point localX(localDimension);
+	  stop += localDimension;
+	  std::copy(x.begin() + start, x.begin() + stop, localX.begin());
+	  const Point localResult(distributionCollection_[i].computeSequentialConditionalPDF(localX));
+	  std::copy(localResult.begin(), localResult.end(), result.begin() + start);
+	  start = stop;
+	} // i
+    } // else
+  return result;
+}
+
+/* Compute the CDF of Xi | X1, ..., Xi-1. x = Xi, y = (X1,...,Xi-1) */
+Scalar BlockIndependentDistribution::computeConditionalCDF(const Scalar x, const Point & y) const
+{
+  const UnsignedInteger conditioningDimension = y.getDimension();
+  if (conditioningDimension >= getDimension()) throw InvalidArgumentException(HERE) << "Error: cannot compute a conditional CDF with a conditioning point of dimension greater or equal to the distribution dimension.";
+  // Special case for no conditioning or independent copula
+  if ((conditioningDimension == 0) || (hasIndependentCopula())) return getMarginal(y.getDimension()).computeCDF(x);
+  // General case
+  UnsignedInteger distributionIndex = 0;
+  UnsignedInteger partialDimension = distributionCollection_[distributionIndex].getDimension();
+  while (partialDimension < conditioningDimension)
+  {
+    ++distributionIndex;
+    partialDimension += distributionCollection_[distributionIndex].getDimension();
+  }
```

Comment:
maybe a small function to get the partial dimension, as this block of code is repeated several times

Reply:
done